lib/models/lookup.js

```diff
@@ -81,7 +81,7 @@ function LookupModelFactory (waterline, Model, assert, Errors, Promise, Constant
 
             switch(dbType) {
                 case 'mongo':
-                    return self.findAndModifyMongo(query, {}, { $set: { node: node }}, options);
+                    return self.findAndModifyMongo(query, {}, { $set: { node: waterline.workitems.mongo.objectId(node)}}, options);
```

Comment:
Moght be worth changing this to a waterline.lookups object instead of waterline.workitems.